pkg/services/sqlstore/dashboard.go

```diff
@@ -68,17 +69,43 @@ func SaveDashboard(cmd *m.SaveDashboardCommand) error {
 			}
 		}
 
-		affectedRows := int64(0)
+		parentVersion := dash.Version
+		version, err := getMaxVersion(sess, dash.Id)
```

Comment:
what was the reason for getMaxVersion? incrementing the current by one should always be safe, as if there is a version mismatch it either generates an error or if overwrite is true, the version will be taken from the current latest version, so dash.Version+1 should always be the same as getMaxVersion